src/main/java/org/spongepowered/api/event/cause/EventContext.java

```diff
@@ -97,6 +97,19 @@ public static Builder builder() {
         return Optional.ofNullable((T) this.entries.get(key));
     }
 
+    @SuppressWarnings("unchecked")
+    public <T> Optional<T> findByName(String possibleName, Class<T> klass) {
+        checkNotNull(possibleName, "String names cannot be null");
+        checkNotNull(klass, "Type class cannot be null");
+        checkArgument(!possibleName.isEmpty(), "String names cannot be empty");
+        final String simplified = possibleName.toLowerCase().replace('_', ' ');
```

Comment:
Just have this get the EventContextKey from the registry by id, check that the key's type matches klass, and then return the get.